src/core/kmod-setup.c

```diff
@@ -45,6 +53,41 @@ static void systemd_kmod_log(
 }
 #endif
 
+static bool has_virtio_rng(void) {
+        char * const paths[] = { (char *)"/sys/devices/pci0000:00", NULL };
+        _cleanup_fts_close_ FTS *fts = NULL;
+
+        fts = fts_open(paths, FTS_NOCHDIR|FTS_NOSTAT, NULL);
+        for (FTSENT *ftsent = fts_read(fts); ftsent != NULL; ftsent = fts_read(fts)) {
+                _cleanup_fclose_ FILE *f = NULL;
+                char alias[2048];
+
+                if (strncmp("modalias", ftsent->fts_name, 8) != 0)
```

Comment:
we prefer "strneq()" for this, in the systemd sources, see CODING_STYLE.

That said, why do you use strncmp() at all here, i.e. why the limitation to 8 chars? You effectively do a prefix match here only, but why?